pkgs/tesla-custom-component/default.nix

```diff
@@ -33,9 +30,13 @@ with home-assistant.python.pkgs; buildHomeAssistantCustomComponent rec {
     teslajsonpy
   ];
 
+  installPhase = ''
+    pypaInstallPhase
+  '';
```

Comment:
@mweinelt just trying out the new API. I wonder if it's worth having an option for `buildHomeAssistantComponent` to keep the original `installPhase`?

Reply:
I guess we could tie this the bespoke installPhase to `format = "other"`.

Reply:
Yes that's probably a good idea

I'll port the rest of my custom components over and see how things look. Perhaps some of them could be good candidates for upstreaming into nixpkgs now.

Reply:
Looking at this more, I think it's probably best not to do this based on `format`- I've found that a bunch of components don't install the right things with their pyproject-based install, so you still end up just copying things from the source directory anyway. It's just good to build the package properly to discover any issues early.

So I'd suggest we just have a way of telling `buildHomeAssistantComponent` not to override `installPhase` via an arg, just in case it's needed (since otherwise people need to dig into the innards of `mkPythonDerivation` to understand what to do in `installPhase`).